amplify/function/BusinessLogic/ResetHackathon/handler.ts

```diff
@@ -216,77 +199,121 @@ export const handler: Handler = async (event) => {
       }
     }
 
-    // Edit start date
-    if (startDate) {
-      console.log("editing start date");
-      const { errors } = await client.graphql({
-        query: updateHackathon,
-        variables: {
-          input: {
-            id: HackathonID,
-            startDate: startDate,
+    // get the current hackathon model data items
+    const { data: hackathonItems } = await client.graphql({
+      query: listHackathons,
+    });
+    const HackathonItems = hackathonItems.listHackathons.items;
+
+    if (!(HackathonItems.length === 0)) {
+      console.log("resetting hackathon");
+      const HackathonID = HackathonItems[0].id;
+      // Edit start date
+      if (startDate) {
```

Comment:
I.e. couldn't we just pass the start date all the time? if it's undefined then the update query just won't edit that field. Since we have only one update method, it will just update anything that is defined.

Reply:
Due to the frontend none of the fields (minus resets) can be undefined but I can make it so that it can be - although well have to say somewhere you can do this

Its called reset hackathon so the user might not even know its also for "editing" a hackathon as well